core/src/main/java/org/apache/accumulo/core/client/AccumuloClient.java

```diff
@@ -99,6 +99,20 @@ BatchScanner createBatchScanner(String tableName, Authorizations authorizations,
   BatchScanner createBatchScanner(String tableName, Authorizations authorizations)
       throws TableNotFoundException;
 
+  /**
+   * Factory method to create a BatchScanner with all of user's authorizations and the number of
+   * query threads configured when AccumuloClient was created. If no query threads were configured,
+   * defaults will be used.
+   *
+   * @param tableName
+   *          the name of the table to query
+   *
+   * @return BatchScanner object for configuring and querying
+   * @throws TableNotFoundException
+   *           when the specified table doesn't exist
+   */
+  BatchScanner createBatchScanner(String tableName) throws TableNotFoundException, AccumuloSecurityException, AccumuloException;
```

Comment:
With the pluggable authorizations, a user could have an infinite set. So, it may not be sufficient to document that they retrieve and submit their authorizations. I think it makes more sense to provide the method. The first pass implementation could get the user authorizations, but later, we'll probably want to pass along a flag that says "don't intersect with user-provided auths" to the visibility filter when it evaluates visibilities.